crypto/stark-verifier-ethereum/contracts/stark_verifier.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity ^0.6.4;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/ConstraintInterface.sol';
+import './public_coin.sol';
+
+
+contract StarkVerifier {
+    using PublicCoin for PublicCoin.Coin;
+    event log_bytes32(bytes32 data);
+
+    // This struct contains all of the componets of the STARK proof.
+    // Please note that any input which would be a 'FieldElement' in rust,
+    // should be the montgomery bytes encoded field element
+    struct StarkProof {
+        // An array with the public inputs to the STARK
+        bytes32[] public_inputs;
+        // An array with the flatened trace table decommitments
+        // For a trace table with n coloums it will be length num_queries*n
+        // and it will be laid out as:
+        // [[query 1 col 1][query 1 col 2]...[query 1 col n]]...[[query q col 1]...[query q col n]]
+        bytes32[] trace_values;
+        // The commitment to those trace values
+        bytes32 trace_root;
```

Comment:
Name is a leaky abstraction: It's a commitment, that it is a merkle *root* is an implementation detail.

Reply:
Same below for constraint and fri